armi/utils/plotting.py

```diff
@@ -1047,3 +1046,237 @@ def makeHistogram(x, y):
         yHistogram[upper] = y[i]
     xHistogram[0] = x[0] / 2.0
     return xHistogram, yHistogram
+
+
+def _makeBlockPinPatches(block, temp):
+    """Return lists of block component patches and corresponding data and names (which relates to material
+    of the component for later plot-coloring/legend) for a single block.
+
+
+    Takes in a block that must have a spatialGrid attached as well as a variable
+    which signifies whether the dimensions of the components are at hot or cold temps.
+    When cold is set to true, you would get the BOL cold temp dimensions.
+
+    Parameters
+    ----------
+    block : block
+
+    temp : boolean
+        cold = true, hot = false
+
+    Return
+    ------
+    patches : list
+        list of patches for block components
+
+    data : list
+        list of the materials these components are made of
+
+    name : list
+        list of the names of these components
+    """
+    patches = []
+    data = []
+    name = []
+    if isinstance(block.spatialGrid, grids.HexGrid):
+        nSides = 6
+        smallestInnerPitch = block.getPitch()
+
+    elif isinstance(block.spatialGrid, grids.ThetaRZGrid):
+        raise ValueError(
```

Comment:
This should actually be a TypeError (https://docs.python.org/3/library/exceptions.html#TypeError)